esp_encrypted_img/include/esp_encrypted_img.h

```diff
@@ -114,12 +115,38 @@ esp_err_t esp_encrypted_img_decrypt_data(esp_decrypt_handle_t ctx, pre_enc_decry
 *
 * @param[in]   ctx   esp_decrypt_handle_t handle
 *
+* @note This API cleans the decrypt handle and return ESP_FAIL if the complete data has not been decrypted. Verify if complete data
+*       has been decrypted using API `esp_encrypted_img_is_complete_data_decrypted` to prevent an early call to this API.
+*
 * @return
-*    - ESP_FAIL    On failure
-*    - ESP_OK
+*    - ESP_FAIL                 On failure
+*    - ESP_ERR_INVALID_ARG      Invalid argument
+*    - ESP_OK                   Success
 */
 esp_err_t esp_encrypted_img_decrypt_end(esp_decrypt_handle_t ctx);
 
+/**
+ * @brief      Checks if the complete data has been decrypted.
+ *
```

Comment:
Shall we add a note here indicating when this API could be invoked and how it could be useful?